Description: Install ld.so to slibdir instead of rtlddir to fix cross builds
 When installing a cross-libc purely for linking purposes but not
 runtime use, the linker shouldn't be installed in rtlddir, since
 we won't actually be USING it, and thus referencing the rtlddir
 path in libc.so ends up blowing up cross-compilers for no reason.
Author: Adam Conrad <adconrad@ubuntu.com>

diff --git a/Makeconfig b/Makeconfig
index 6618987..b9f8680 100644
--- a/Makeconfig
+++ b/Makeconfig
@@ -161,7 +161,7 @@ inst_slibdir = $(install_root)$(slibdir)
 ifndef rtlddir
 rtlddir = $(slibdir)
 endif
-inst_rtlddir = $(install_root)$(rtlddir)
+inst_rtlddir = $(install_root)$(slibdir)
 
 # Prefix to put on files installed in $(libdir).  For libraries `libNAME.a',
 # the prefix is spliced between `lib' and the name, so the linker switch
diff --git a/Makerules b/Makerules
index 92c2872..cb45f90 100644
--- a/Makerules
+++ b/Makerules
@@ -960,7 +960,7 @@ $(inst_libdir)/libc.so: $(common-objpfx)format.lds \
 	 cat $<; \
 	 echo 'GROUP ( $(slibdir)/libc.so$(libc.so-version)' \
 	      '$(libdir)/$(patsubst %,$(libtype.oS),$(libprefix)$(libc-name))'\
-	      ' AS_NEEDED (' $(rtlddir)/$(rtld-installed-name) ') )' \
+	      ' AS_NEEDED (' $(slibdir)/$(rtld-installed-name) ') )' \
 	) > $@.new
 	mv -f $@.new $@
 
